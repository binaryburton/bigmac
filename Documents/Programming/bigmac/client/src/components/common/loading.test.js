import React from "react";
import ReactDOM from "react-dom";
import Loading from "./loading";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders withougt crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Loading></Loading>, div);
});

it("renders loading screen correctly", () => {
  const { getByTestId } = render(<Loading></Loading>);
  expect(getByTestId("loading")).toBeInTheDocument();
});

it("matches snapshot", () => {
  const tree = renderer.create(<Loading></Loading>).toJSON();
  expect(tree).toMatchSnapshot();
});
