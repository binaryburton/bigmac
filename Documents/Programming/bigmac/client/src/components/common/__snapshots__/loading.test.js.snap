// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshop 1`] = `
<div
  data-testid="loading"
  style={
    Object {
      "color": "green",
    }
  }
>
  <b>
    <h1>
      LOADING...
    </h1>
    <h2>
      LOADING...
    </h2>
    <h3>
      LOADING...
    </h3>
    <h4>
      LOADING...
    </h4>
    <h5>
      LOADING...
    </h5>
    <h6>
      LOADING...
    </h6>
  </b>
</div>
`;

exports[`matches snapshot 1`] = `
<div
  data-testid="loading"
  style={
    Object {
      "color": "green",
    }
  }
>
  <b>
    <h1>
      LOADING...
    </h1>
    <h2>
      LOADING...
    </h2>
    <h3>
      LOADING...
    </h3>
    <h4>
      LOADING...
    </h4>
    <h5>
      LOADING...
    </h5>
    <h6>
      LOADING...
    </h6>
  </b>
</div>
`;
